### PR TITLE
Feat/filter media src

### PR DESCRIPTION
**Pull request by markkelnar**
For responsive images in srcset, replace domain to WP backend url.

### CODE REVIEW

File: plugins/wpe-headless/includes/replacement/callbacks.php [hunk 1]
```diff
@@ -52,11 +51,42 @@ function wpe_headless_content_media_replacement( $content ) {
...
+/**
+ * Callback for WordPress 'the_content' filter to replace paths to media.
+ *
+ * @param array $sources One or more arrays of source data to include in the 'srcset'.
+ *
+ * @return string One or more arrays of source data.
+ */
+function wpe_headless_content_media_srcset_replacement( $sources ) {
+	if ( ! wpe_headless_domain_replacement_enabled() ) {
+		return $content;
```
Comment (markkelnar) [inline plugins/wpe-headless/includes/replacement/callbacks.php:72]: Look at that bug.  Darn it.

**Review by rfmeier — APPROVED** [review]
Looks good. I was able to replicate locally. I like the new use of `preg_replace()` 👍

**Review by rfmeier — APPROVED** [review]

File: plugins/wpe-headless/includes/settings/callbacks.php [hunk 2]
```diff
@@ -286,6 +287,12 @@ function wpe_headless_display_enable_disable_fields() {
 			<input type="checkbox" id="enable_redirects" name="wpe_headless[enable_redirects]" value="1" <?php checked( $enable_redirects ); ?> />
 			<?php esc_html_e( 'Enable public route redirects', 'wpe-headless' ); ?>
 		</label>
+		<br />
+
+		<label for="enable_image_source">
+			<input type="checkbox" id="enable_image_source" name="wpe_headless[enable_image_source]" value="1" <?php checked( $enable_image_source ); ?> />
+			<?php esc_html_e( 'Enable media images served from WP endpoint', 'wpe-headless' ); ?>
```
Comment (nickcernis) [inline plugins/wpe-headless/includes/settings/callbacks.php:294]: Would this be more clear for the checkbox label? cc @trevanhetzel 

```suggestion
			<?php esc_html_e( 'Use the WordPress domain for media URLs in post content', 'wpe-headless' ); ?>
```

**Review by nickcernis — APPROVED** [review]
Working well for me too.

In the post:

Screenshot 2021-01-27 at 11 36 36

In the frontend app source:

Screenshot 2021-01-27 at 11 36 48

I suggested a tweak to the checkbox label but it doesn't block merge.

**Pull request merged**